recipes/openscenegraph/all/patches/0009-replace-auto-ptr-in-plugins.patch

```diff
@@ -0,0 +1,75 @@
+diff --git a/src/osgPlugins/dae/ReaderWriterDAE.cpp b/src/osgPlugins/dae/ReaderWriterDAE.cpp
+index fc1a448d4..3b883f19a 100644
+--- a/src/osgPlugins/dae/ReaderWriterDAE.cpp
++++ b/src/osgPlugins/dae/ReaderWriterDAE.cpp
+@@ -32,7 +32,7 @@
+ 
+ #define SERIALIZER() OpenThreads::ScopedLock<OpenThreads::ReentrantMutex> lock(_serializerMutex)
+ 
+-#if  __cplusplus > 199711L
++#if ((defined(_MSVC_LANG) && _MSVC_LANG > 199711L) || __cplusplus > 199711L)
+     #define smart_ptr std::unique_ptr
+ #else
+     #define smart_ptr std::auto_ptr
+diff --git a/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp b/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
+index 69826c456..9bba5532a 100644
+--- a/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
++++ b/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
+@@ -10,6 +10,11 @@
+ 
+ #define STREAM_TIMEOUT_IN_SECONDS_TO_CONSIDER_IT_DEAD   10
+ 
++#if ((defined(_MSVC_LANG) && _MSVC_LANG > 199711L) || __cplusplus > 199711L)
++    #define smart_ptr std::unique_ptr
++#else
++    #define smart_ptr std::auto_ptr
++#endif
```

Comment:
Would it be possible to use a `using` statement here instead of a macro when using C++ 11 and newer? I don't think a templated `typedef` is possible for earlier C++ standards.

```suggestion
+#if ((defined(_MSVC_LANG) && _MSVC_LANG > 199711L) || __cplusplus > 199711L)
+    template <typename T> using smart_ptr = std::unique_ptr<T>;
+#else
+    #define smart_ptr std::auto_ptr
+#endif
```

Reply:
Yes, should work :)